### PR TITLE
Initial release

### PR DESCRIPTION
**Pull request by mrjackv**
This is the first release of the vscode extension.
Functionality-wise the plugins offers references, definitions and folding ranges.
Development-wise there are linting rules, unit tests and documentation in place.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# LLVM IR Language Suppport
```
Comment (aleclearmind) [inline README.md]:
```suggestion
# LLVM IR Language Support
```

Can we have build and install instructions for dumb people like me?

Comment (mrjackv) [reply README.md]: * Typo fixed
* Added `COMPILE.md` with compilation instructions

**Comment by aleclearmind** [comment]
I've tried, super-cool! Great job!
Some notes:
* [ ] The `%`, `@` are not ctrl-clickable. This makes it hard to navigate through `%5` (you can only click on the `5`).
* [ ] The underlying when you press `Ctrl` seems wrong, it splits over `.` while it shouldn't.
  image
  `Ctrl+click` works properly though!
* [ ] References do not work %struct.PlainMetaAddress
* [ ] How much context on hover?
* [ ] Arguments seem not to work (maybe %0?)
* [ ] References to attribute sets (#0)
* [ ] Clickability of non-literal/digits (can't click on @ and .)

File: src/web/llvmir/regexp.ts [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        ${globalVarFrag}|           # Global Identifiers
+        ${allLocalVarFrag}|         # Local variables
+        ${attributeGroupFrag}|      # Attributes
+        ${metadataFrag}             # Metadata
+    )`);
+
+    // Regexes
+
+    // Generic identifier regex, without named capture
+    // Used with getWordRangeAtPosition
```
Comment (aleclearmind) [inline src/web/llvmir/regexp.ts]: Let's be coherent with comment prefixes. Either `//` or `/**/`

File: LICENSE.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2022 Alessandro Di Federico
```
Comment (aleclearmind) [inline LICENSE.md]: rev.ng Labs S.r.l.

File: COMPILE.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
+## Compilation Instructions
```
Comment (aleclearmind) [inline COMPILE.md]: * [ ] Merge with `README.md`
* [ ] Add testing instructions

**Review by aleclearmind — CHANGES_REQUESTED** [review]

File: COMPILE.md (hunk 4 above)
Comment (mrjackv) [reply COMPILE.md]: The problem is that the README is what gets rendered as the text in the marketplace and in the extension's description.
That's the reason it is a separate file right now

Comment (aleclearmind) [reply COMPILE.md]: OK, then this file should be called `INSTALL.md`.

**Comment by mrjackv** [comment]
I've fixed all the issued outline in [this comment](https://github.com/revng/vscode-llvm-ir/pull/1#issuecomment-1058995838)

File: COMPILE.md (hunk 4 above)
Comment (mrjackv) [reply COMPILE.md]: Fixed

**Comment by aleclearmind** [comment]
If I click on "Find references" for `%0` on `handle_sigsegv` I get *all* the references of all the `%0`s instead of only those in `handle_sigsegv`.
All the other limitations seem fixed.

**Comment by mrjackv** [comment]
References are now fixed, there was a badly-expressed condition that caused all the variables to be put in the global namespace.

**Comment by aleclearmind** [comment]
Merged in 392422f. Thanks!
Great job.

**Pull request closed without merging**